test/integration/api/odata-entities.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const { testService } = require('../setup');
+const testData = require('../../data/xml');
+const { identity } = require('ramda');
+const { exhaust } = require('../../../lib/worker/worker');
+const { v4: uuid } = require('uuid');
+
+describe('api: /datasets/:name.svc', () => {
+  describe('GET /Entities', () => {
+
+    /* eslint-disable no-await-in-loop*/
+    const createSubmissions = async (user, container, count = 1) => {
+      for (let i = 0; i < count; i += 1) {
+        await user.post('/v1/projects/1/forms/simpleEntity/submissions')
+          .send(testData.instances.simpleEntity.one
+            .replace(/one/g, `submission${i}`)
+            .replace('uuid:12345678-1234-4123-8234-123456789abc', uuid()))
+          .set('Content-Type', 'application/xml')
+          .expect(200);
+
+        await user.patch(`/v1/projects/1/forms/simpleEntity/submissions/submission${i}`)
+          .send({ reviewState: 'approved' })
+          .expect(200);
+
+      }
+      await exhaust(container);
+    };
+    /* eslint-enable no-await-in-loop*/
+
+    it('should return all entities', testService(async (service, container) => {
+      const asAlice = await service.login('alice', identity);
```

Comment:
I just merged a PR that makes the test callback optional. So you can now remove `identity` here (if you rebase)!